### ISSUE TITLE
NODE_PATH matches the npm root - Unable to find the npm root, something went wrong.

### ISSUE THREAD

**Issue by nandaccio** [yeoman/doctor#59]
I am unable to run yo, so I ran "yo doctor" and I am getting the error message below.

Yeoman Doctor
Running sanity checks on your system

√ No .bowerrc file in home directory
√ Global configuration file is valid
× NODE_PATH matches the npm root

Unable to find the npm root, something went wrong.
Try to execute npm -g root --silent on your command-line

√ No .yo-rc.json file in home directory
√ Node.js version
√ npm version
√ yo version

Found potential issues on your machine :(

NOTES:
* NODE_PATH is set to c:\Program Files\nodejs\node_modules\
* npm -g root --silent returns: C:\Program Files\nodejs\node_modules
* Node.js version: 17.3.0
* I ran "yo doctor" both as a regular user and as an administrator, same issue
* yo --version: 4.3.0

![image](https://user-images.githubusercontent.com/13116598/146962455-0721880c-604b-41c6-bc6e-54eed175e4b6.png)

**Comment by appsparkler** [comment]
Placing the generator `index.js` file inside `generators/app` instead of 

```js
// ./generator/app/index.js
var Generator = require('yeoman-generator');

// Try this sample generator
module.exports = class extends Generator {
    constructor(args, opts) {
        // Calling the super constructor is important so our generator is correctly set up
        super(args, opts);
    }
    method1() {
        this.fs.write(this.destinationPath('index.js'), 'const foo = 1;');
    }
};
```

Now, you might still have an error with `yo doctor`; however, you'll have a working generator.

A better approach would be is to use Yeoman's [generator-generator](https://github.com/yeoman/generator-generator) as a starter kit and then build on top of that.

**Comment by JoshuaKGoldberg** [comment]
Indeed, seems like a userland issue with `NODE_PATH` vs. `npm -g root`. I think `yo doctor` is right to report here. Given that it's been a few years since anybody posted, I'll go ahead and close this out. But someone should definitely yell at me if there's really an issue in the tool. Cheers! 🎩

**Comment by johnholliday** [comment]
This appears to be a real issue.  I'm not building a custom version - fails out of the box.
